libs/MeshKernel/include/MeshKernel/Mesh2D.hpp

```diff
@@ -72,9 +72,21 @@ namespace meshkernel
         /// @brief Default constructor
         Mesh2D();
 
+        /// @brief Delete assignment operator
+        Mesh2D& operator=(const Mesh2D& mesh2D) = delete;
+
+        /// @brief Delete move assignment operator
+        Mesh2D& operator=(Mesh2D&& mesh2D) = delete;
+
+        /// @brief Copy constructor taking a mesh2D
+        Mesh2D(const Mesh2D& mesh2D) = delete;
+
+        /// @brief Move constructor taking a mesh2D
+        Mesh2D(Mesh2D&& mesh2D) = delete;
+
```

Comment:
As in Mesh1D, check if this needed given that the base is non-copyable and non-movable